qml/Ping360Visualizer.qml

```diff
@@ -0,0 +1,120 @@
+import QtGraphicalEffects 1.0
+import QtQuick 2.7
+import QtQuick.Controls 2.2
+import QtQuick.Controls 1.4 as QC1
+import QtQuick.Layouts 1.3
+import QtQuick.Shapes 1.0
+import Qt.labs.settings 1.0
+import WaterfallPlot 1.0
+import PolarPlot 1.0
+
+import SettingsManager 1.0
+import StyleManager 1.0
+
+Item {
+    id: visualizer
+    property alias waterfallItem: waterfall
+    property var protocol
+
+    onWidthChanged: {
+        if(chart.Layout.minimumWidth === chart.width) {
+            waterfall.parent.width = width - chart.width
+        }
+    }
+
+    function draw(points, confidence, initialPoint, length, distance) {
+        waterfall.draw(points, confidence, initialPoint, length, distance)
+        chart.draw(points, length + initialPoint, initialPoint)
+    }
+
+    function setDepth(depth) {
+        readout.value = depth
+    }
+
+    function setConfidence(perc) {
+        readout.confidence = perc
+    }
+
+    QC1.SplitView {
+        orientation: Qt.Horizontal
+        anchors.fill: parent
+
+        Item {
+            Layout.fillHeight: true
+            Layout.fillWidth: true
+
+            PolarPlot {
+                id: waterfall
+                height: Math.min(parent.height, parent.width)
+                width: height
+                anchors.centerIn: parent
+
+                Shape {
```

Comment:
TODO: this should be shaped like a 'pie slice' instead of a rectangle, and the angular resolution should be quantized to match the sensor (400 steps).